platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/ShadowGraph.java

```diff
@@ -179,6 +179,16 @@ public synchronized void initFromEvents(final List<EventImpl> events, final long
                 oldestGeneration);
     }
 
+    /**
+     * Define the starting generation for the shadowgraph, it will not keep any events older than this
+     * @param generation the starting generation
+     */
+    public void startFromGeneration(final long generation) {
+        expireBelow = generation;
+        oldestGeneration = generation;
+        logger.info(STARTUP.getMarker(), "Shadow graph starting from generation {}", generation);
```

Comment:
Perhaps this method should throw if it is called and there is anything in the shadowgraph already.

Reply:
generally, this is called on either a new instance or a cleared instance, so there should be no events stored when this is called. I could add a `clear()` call to the beginning of the method to be safe?